builder/qemu/builder.go

```diff
@@ -118,6 +118,7 @@ func (b *Builder) Run(ctx context.Context, ui packersdk.Ui, hook packersdk.Hook)
 			QMPSocketPath: b.config.QMPSocketPath,
 		},
 		&stepTypeBootCommand{},
+		&stepTypeBootSteps{},
```

Comment:
As mentioned below, if we make the two mutually-exclusive, I would think we can manage both in the BootCommand step alone, no need for a new step to handle them separately. We could adopt the logic you implemented as common, and the only change would be to select the appropriate boot commands to execute.

Reply:
Ok, I can look into doing that.